### PR TITLE
Internal login start endpoint

### LINKED ISSUE #2424: Provide an endpoint to initiate admin login
We need an endpoint to initiate the admin login flow. This will set the state cookie and redirect the user to FxA using the admin FxA client id.

Supports #2385.

### PR DESCRIPTION
**Pull request by mstriemer**
This endpoint will kick off the login flow by setting `fxa_state` and redirecting to FxA.

Fixes #2424.

### CODE REVIEW

File: docs/topics/api/auth_internal.rst [hunk 1]
```diff
@@ -26,6 +26,13 @@ responses of the following endpoints:
     * ``/api/v3/accounts/register/``
     * ``/api/v3/accounts/authenticate/``
 
+A JWT may also be obtained through the JSON API as outlined in the
+:ref:`internal login JSON API <internal-login-json-api>` section.  This is only
```
Comment (EnTeQuAk) [inline docs/topics/api/auth_internal.rst]: There's a double whitespace before `This is only`

File: settings.py [hunk 2]
```diff
@@ -98,6 +98,16 @@
     'redirect_url': 'http://olympia.dev/api/v3/accounts/authorize/',
     'scope': 'profile',
 }
+ADMIN_FXA_CONFIG = {
+    'client_id': '0f95f6474c24c1dc',
+    'client_secret':
+        'ca45e503a1b4ec9e2a3d4855d79849e098da18b7dfe42b6bc76dfed420fc1d38',
```
Comment (EnTeQuAk) [inline settings.py]: Is this a test token?

File: src/olympia/internal_tools/tests/test_views.py [hunk 3]
```diff
@@ -151,3 +163,73 @@ def test_pagination_sort_and_query(self):
...
+        with mock.patch('olympia.internal_tools.views.generate_fxa_state',
+                        lambda: 'arandomstring'):
+            response = self.client.get(self.url)
+        assert response.status_code == 302
+        url = urlparse.urlparse(response['location'])
+        assert '{scheme}://{netloc}{path}'.format(
+            scheme=url.scheme,
+            netloc=url.netloc,
+            path=url.path,
+        ) == 'https://accounts.firefox.com/v1/authorization'
```
Comment (EnTeQuAk) [inline src/olympia/internal_tools/tests/test_views.py]: I'm really a fan of

```python
result = '{scheme}://{netloc}{path}'.format(url.scheme, url.netloc, url.path)
assert result == 'https://accounts.firefox.com/v1/authorization'
```

But that's just a nitpick and not too important.

**Comment by EnTeQuAk** [comment]
r+wc

File: settings.py (hunk 2 above)
Comment (mstriemer) [inline settings.py]: It's the one for addons-frontend development. It only works on `http://localhost:3000`. We do something similar for the regular FxA setup that only works on `http://olympia.dev`.

Comment (mstriemer) [inline settings.py]: This isn't included on any other environment either (or won't be, I guess).

**Pull request merged**